packages/deepbook/sources/balance_manager.move

```diff
@@ -168,6 +224,61 @@ public fun revoke_trade_cap(
     balance_manager.allow_listed.remove(trade_cap_id);
 }
 
+public fun set_referral(
+    balance_manager: &mut BalanceManager,
+    trade_cap: &TradeCap,
+    referral_cap: &DeepBookReferral,
+    ctx: &mut TxContext,
+) {
+    balance_manager.validate_trader(trade_cap);
+    let ref_key = constants::referral_df_key();
+    let referral_id = referral_cap.id.to_inner();
+    let current_fees = if (balance_manager.id.exists_(ref_key)) {
```

Comment:
instead of transferring to the new referrer, can we require a claim before reseting or maintain the previous balance?